powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java

```diff
@@ -2001,6 +2001,13 @@ private ActivationRecovery createRecoveryCodeForActivation(ActivationRecordEntit
                 throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_REQUEST);
             }
 
+            // Check if protocol is POWERAUTH
+            if (!Protocols.POWERAUTH.toString().equals(activationEntity.getProtocol())) {
+                logger.warn("Invalid protocol in method createRecoveryCodeForActivation");
```

Comment:
Nitpick: Personally, instead of the comment, I prefer a method. Moreover, `toString()` is fragile. I would call `POWERAUTH.getProtocol()` or even introduce a new method `POWERAUTH.isProtocol(String)` (or some better name).

Reply:
Good point, I will introduce methods for these checks.